### ISSUE TITLE
Download Support for Region Restricted apps

### ISSUE THREAD

**Issue by khaleel-git** [NoMore201/googleplay-api#157]
I wonder if there's a way with this repository (@NoMore201) to download region restricted apps? If i know the region where the specific apk is supported, i can download it with that specific location configurations but the problem is, how do we know the specific app (for example: https://play.google.com/store/apps/details?id=com.sonyliv) is supported in which specific country? or is blocked on which countries?  

Please let me know if its possible ...

**Comment by 89z** [comment]
That one works with US, not sure with other countries:

~~~
PS C:\> googleplay.exe -a com.sonyliv -v 9605
POST https://android.googleapis.com/auth
GET https://play-fe.googleapis.com/fdfe/delivery?doc=com.sonyliv&vc=9605
GET https://play.googleapis.com/download/by-token/download?token=AOTCm0RC9iA3MGRCa4VP23nvucIUMDks_z19DsHUXWVJk1koLfLyr6SxPKDBd__FvVMf0SZr_Xp1_l0wM10vIzN_DE_08dUeKFPiQxnrL-1TL4ZzBSDwaUKEWKD7xLeL9lPrsuiBfxzjxR2BVHrlJYxJ3Y5DPkDLaBvRJXhJj8K0Y67OCQ5anJdKJWDxI1osBvjABrtrfIF1H2QL729BRc4i5LZ4FLLWXOAhXyDIPowzpWif-Cfn_70zzs0Ao6RYDIwvadsAxbWatR5fsgl1r0Nx7uIermS--VrdglVTH-Adelxhf92LSFGXutQliBMNWXfdkQPJAAONQ2if1j4IjBOFPGLyBXXhFZMH8B7KU0ukjCtgJ4jxQtSGFv50oXqFqFvDT3130MXaIGcX&cpn=itTj2Dk-SDjdAo4C
12.4%   3.818 MB        7.613 MB/s
~~~

**Comment by khaleel-git** [comment]
RequestError: 'Error retrieving information from server. DF-DFERH-01' with @[googleplay-api](https://github.com/NoMore201/googleplay-api/) 
**My Code:** 
api = GooglePlayAPI("en_US", timezone = "America/Los_Angeles", device_codename = "manta") #bacon, fs454, manta
api.login(email = mail, password = passwd)

app_id = "com.sonyliv"
docId = app_id
download = api.download(docId, expansion_files=True)
with open(download['docId'] + '.apk', 'wb') as first:
 for chunk in download.get('file').get('data'):
  first.write(chunk)

**Comment by 89z** [comment]
Try just the `details` function first, see if that works

**Comment by khaleel-git** [comment]
details function working fine, couldn't download apk.

**Comment by 89z** [comment]
Since youre not physically in the US, you would need to change your location:

https://support.google.com/googleplay/answer/7431675

or use a proxy. Note, you can only change your location once per year. If you find some other way, let me know.

**Comment by khaleel-git** [comment]
> 

Yes, I am currently researching about this. Can u seggest me a solution with proxy? I mean how can i download restriced apk by applying proxy?

**Comment by khaleel-git** [comment]
> Sorry, I am not good with proxy. I think free proxy are bad quality, so you probably need to find paid proxy. Good luck, or maybe you can find another solution.

I think it also require a separate google account for each proxy country.

**Comment by Exadra37** [comment]
To download a restricted APK you may ned to first use a compatible device for that country and that is supported by that APK. Go to device.properties and which device uses a country not restricted by the APK and that uses an Android version supported by the APK. If a profile doesn't exist then you need to create a new one.

The bash script I use to update the `device.properties` file:

```bash
#!/bin/bash

# REPO_SRC="https://github.com/yeriomin/play-store-api"
REPO_SRC="https://gitlab.com/AuroraOSS/gplayapi.git"
REPO_LOCAL="/tmp/psapi"
RES_DIR="${REPO_LOCAL}/src/main/resources"

DEVS_FILE="./gpapi/device.properties"

command -v git >/dev/null 2>&1 || { echo "git not installed"; exit 1; }

if [ ! -d "./gpapi" ]; then
        echo "No gpapi dir found! Make sure you're in googleplay-api root dir"
        exit 1
fi

mkdir -p $REPO_LOCAL

echo "==> Cloning play-store-api repo into $REPO_LOCAL"
git clone $REPO_SRC $REPO_LOCAL

# clean device.properties file
echo "" > $DEVS_FILE

for dev in `ls $RES_DIR`; do
        FILE="$RES_DIR/$dev"
        NAME=`echo $dev | sed -e "s/\(.*\).properties/\1/"`
        echo "==> appending device data for $NAME"
        echo -e "\n[$NAME]" >> $DEVS_FILE
        cat $FILE >> $DEVS_FILE
        echo "" >> $DEVS_FILE
done

# cleanup
echo "==> Cleanup"
rm -rf $REPO_LOCAL

```

I can download some restricted APKs with the correct device properties and OS without using a proxy its necessary to use one.

**Comment by ghost** [comment]
above comment is not true. you need to change the country in your Google account settings:

https://github.com/4cq2/googleplay/blob/main/blog/2023-02/geo-blocking.md

unless I see proof otherwise, I am not accepting this claim. provide an example app, and proof of your current region, otherwise the claim is suspect.
